lib/glimesh/streams.ex

```diff
@@ -108,4 +108,58 @@ defmodule Glimesh.Streams do
   def count_following(user) do
     Repo.one!(from f in Followers, select: count(f.id), where: f.user_id == ^user.id)
   end
+
+  alias Glimesh.Streams.StreamMetadata
+
+  def change_title(streamer, title) do
+    attrs = %{
+      stream_title: title
+    }
+    streamerMeta = get_metadata_from_streamer(streamer)
+
+    {:ok, results} =
+      %StreamMetadata{
+        streamer: streamer
+      }
+      |> Map.merge(streamerMeta)
+      |> StreamMetadata.changeset(attrs)
+      |> Repo.update()
+
+    update_data = %{title: results.stream_title, streamer_username: streamer.username}
+    broadcast({:ok, update_data}, :update_title)
+    {:ok, results}
+  end
+
+  def create_metadata(streamer) do
+    attrs = %{
+      stream_title: "My first stream!"
+    }
+    results =
+      %StreamMetadata{
+        streamer: streamer
+      }
+      |> StreamMetadata.changeset(attrs)
+      |> Repo.insert()
+  end
+
+  def get_metadata_from_streamer(streamer) do
+    data = Repo.get_by(StreamMetadata, streamer_id: streamer.id)
+    if !data do
+      create_metadata(streamer)
+      get_metadata_from_streamer(streamer)
+    else
+      data
+    end
+  end
+
+  def subscribe_metadata do
+    Phoenix.PubSub.subscribe(Glimesh.PubSub, "streams")
```

Comment:
If we namespace `streams:user_id` we don't have to do any streamer checks after this. Not sure the implications of so many channels though?

Reply:
We can wait and do this refactor later though, since chat / view count suffer from it right now too. We want to standardize this for the API so I can list to the right channels...